MiBench/automotive/basicmath/src/basicmath_large.c

```diff
@@ -89,35 +99,42 @@ int main(void)
   }
 
 
+
   printf("********* INTEGER SQR ROOTS ***********\n");
-  /* perform some integer square roots */
-  for (i = 0; i < 100000; i+=2)
-    {
-      usqrt(i, &q);
-			// remainder differs on some machines
-     // printf("sqrt(%3d) = %2d, remainder = %2d\n",
-     printf("sqrt(%3d) = %2d\n",
-	     i, q.sqrt);
-    }
-  printf("\n");
+  
+  start = clock();
+  for (i = 0; i < 100000; i += 2)
+  {
+    usqrt(i, &q);
+    // remainder differs on some machines
+    printf("sqrt(%3d) = %2d\n", i, q.sqrt);
+  }
+  
+  end = clock();
+  cpu_time_used = ((double) (end - start)) / CLOCKS_PER_SEC;
+ 
+
+ printf("\n");
   for (l = 0x3fed0169L; l < 0x3fed4169L; l++)
     {
 	 usqrt(l, &q);
 	 //printf("\nsqrt(%lX) = %X, remainder = %X\n", l, q.sqrt, q.frac);
 	 printf("sqrt(%lX) = %X\n", l, q.sqrt);
     }
 
-
   printf("********* ANGLE CONVERSION ***********\n");
-  /* convert some rads to degrees */
-/*   for (X = 0.0; X <= 360.0; X += 1.0) */
+  
+  start = clock();
   for (X = 0.0; X <= 360.0; X += .001)
     printf("%3.0f degrees = %.12f radians\n", X, deg2rad(X));
   puts("");
-/*   for (X = 0.0; X <= (2 * PI + 1e-6); X += (PI / 180)) */
+
   for (X = 0.0; X <= (2 * PI + 1e-6); X += (PI / 5760))
     printf("%.12f radians = %3.0f degrees\n", X, rad2deg(X));
   
-  
+  end = clock();
+  cpu_time_used = ((double) (end - start)) / CLOCKS_PER_SEC;
```

Comment:
Will this not overwrite the previous assignment to `cpu_time_used`?

Reply:
Yes Mats.
I am adding total_cpu_time += cpu_time_used; in order for it to not overlap.